### PR TITLE
fix: add postgres columns validations and connector test

### PR DESCRIPTION
**Pull request by hoangnh93**

### CODE REVIEW

File: dozer-ingestion/src/connectors/postgres/connection/validator.rs [hunk 1]
```diff
@@ -176,17 +177,50 @@ fn validate_tables(
...
+            .iter()
+            .find(|x| x.table_name == table_name)
+            .unwrap()
+            .clone()
+            .columns;
+
+        if let Some(column_info) = columns {
+            let column_result = client
+                .query(
+                    "SELECT column_name FROM information_schema.columns WHERE table_name = ($1)",
```
Comment (karolisg) [inline dozer-ingestion/src/connectors/postgres/connection/validator.rs:195]: It doesn't look efficient to run query on every table separately

**Review by karolisg — CHANGES_REQUESTED** [review]

File: dozer-ingestion/src/connectors/postgres/snapshotter.rs [hunk 2]
```diff
@@ -112,3 +112,344 @@ impl<'a> PostgresSnapshotter<'a> {
...
+            let (ingestor, mut _iterator) = Ingestor::initialize_channel(ingestion_config);
+
+            let snapshotter = PostgresSnapshotter {
+                tables: Some(tables.clone()),
+                conn_config: config,
+                ingestor: &ingestor,
+                connector_id: connector.id,
+            };
+
+            let actual = snapshotter.sync_tables(Some(input_tables));
```
Comment (karolisg) [inline dozer-ingestion/src/connectors/postgres/snapshotter.rs:213]: `sync_tables` no longer return tables. In this test, you need to verify what messages ingestor gets.

**Review by karolisg — CHANGES_REQUESTED** [review]

File: dozer-ingestion/src/connectors/postgres/connection/validator.rs (hunk 1 above)
Comment (hoangnh93) [reply dozer-ingestion/src/connectors/postgres/connection/validator.rs]: updated!

File: dozer-ingestion/src/connectors/postgres/snapshotter.rs (hunk 2 above)
Comment (hoangnh93) [reply dozer-ingestion/src/connectors/postgres/snapshotter.rs]: updated!

**Review by karolisg — APPROVED** [review]

**Pull request merged**